src/app/modules/main/profile_section/profile/models/showcase_preferences_generic_model.nim

```diff
@@ -0,0 +1,71 @@
+import NimQml, tables, strutils, sequtils, json
+
+import app_service/service/profile/dto/profile_showcase_preferences
+
+type
+  ShowcasePreferencesGenericItem* = object of RootObj
+    showcaseKey*: string
+    showcaseVisibility*: ProfileShowcaseVisibility
+
+proc initShowcasePreferencesGenericItem*(key: string, visibility: ProfileShowcaseVisibility, order: int): ShowcasePreferencesGenericItem =
+  result = ShowcasePreferencesGenericItem()
+
+  result.showcaseKey = key
+  result.showcaseVisibility = visibility
+
+type
+  ModelRole {.pure.} = enum
```

Comment:
This model won't have the `showcasePosition`?

Reply:
i'll add it now